compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -446,10 +450,69 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                 }
             }
         }
+
         err.span_label(decl_span, "binding declared here but left uninitialized");
+        if show_assign_sugg {
+            self.suggest_assign_rvalue(&mut err, moved_place, &name, decl_span);
+        }
         err
     }
 
+    fn suggest_assign_rvalue(
+        &self,
+        err: &mut Diagnostic,
+        moved_place: PlaceRef<'tcx>,
+        name: &str,
+        decl_span: Span,
+    ) {
+        let ty = moved_place.ty(self.body, self.infcx.tcx).ty;
+        debug!("ty: {:?}, kind: {:?}", ty, ty.kind());
+
+        let initilize_msg = match ty.kind() {
+            ty::Array(_, n) => format!("[val; {}]", n),
+            ty::Int(_) | ty::Uint(_) => format!("0"),
+            ty::Float(_) => format!("0.0"),
+            ty::Bool => format!("false"),
+            ty::Never | ty::Error(_) => "".to_string(),
+            ty::Adt(def, _substs) => {
+                if format!("{:?}", def).starts_with("std::vec::Vec") {
```

Comment:
Any better solution for checking whether it's a `vec`?

Reply:
`Vec` is a diagnostic item, so you can use `def.did == tcx.get_diagnostic_item(sym::Vec)`.